circom/aes256.go

```diff
@@ -2,28 +2,25 @@ package circom
 
 import (
 	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark/std/math/uints"
 )
 
 type AES256Wrapper struct {
-	Plain  [16]frontend.Variable
-	Key    [32]frontend.Variable
-	Cipher [16]frontend.Variable `gnark:",public"`
+	Plain  [16]uints.U8
+	Key    [32]uints.U8
+	Cipher [16]uints.U8 `gnark:",public"`
```

Comment:
Are you sure it's safe to use?

> // NB! The package is still work in progress. The interfaces and implementation
// details most certainly changes over time. We cannot ensure the soundness of
// the operations.

The major logic is implemented, but there's still a lot of TODOs:
> // TODO: if internal then enforce range check!
// TODO: all operations can take rand linear combinations instead. Then instead
// of one check can perform multiple at the same time.
// TODO: implement versions which take multiple inputs. Maybe can combine multiple together
// TODO: instantiate tables only when we first query. Maybe do not need to build!
...

Reply:
This is an introduction to the gnark library：https://github.com/consensys/gnark?tab=readme-ov-file#audits
If worry about the security and stability of the library, we can discuss it in discord

Reply:
The comments and to-do list have been there since `0.9.0` from last year, and didn't change a lot. Now it is `0.11.0`.

And it is the Gnark team that recommend us to use `uint.U8` for AES crypto in circuits, otherwise there is no other way to implement it.